app/questionnaire/questionnaire_store_updater.py

```diff
@@ -298,16 +327,36 @@ def update_answers(
 
             answer_updated = self._update_answer(answer_id, list_item_id, answer_value)
             if answer_updated:
-                self._capture_dependencies_for_answer(answer_id)
+                self._capture_section_dependencies_for_answer(answer_id)
+                self._capture_block_dependencies_for_answer(answer_id)
 
-    def update_progress_for_dependant_sections(self) -> None:
+    def update_progress_for_dependent_sections(self) -> None:
         """Removes dependent blocks from the progress store and updates the progress to IN_PROGRESS.
+        Section progress is not updated for the current location as it is handled by `handle_post` on block handlers."""
 
-        Section progress is not updated for the current location as it is handled by `handle_post` on block handlers.
+        self._remove_dependent_blocks_and_capture_dependent_sections()
+
+        for section in self.dependent_sections:
+            is_path_complete = section.is_complete
+
+            if (
+                section.is_complete is None
+                and section.section_id not in self.started_section_keys()
```

Comment:
This would always be type. `started_section_keys` are tuples of sections keys, whereas section_id is a string. You need to check section key is not in `started_section_keys`.